### ISSUE TITLE
A11y Audit: [calcite-panel] - inactive elements should not be focusable, remove `tabindex` from section tag

### ISSUE THREAD

**Issue by caripizza** [Esri/calcite-components#3379]
### Issue Summary

Panel's "content-container" section - which houses its default slotted content - has [`tabindex="0"`](https://github.com/Esri/calcite-components/blob/master/src/components/calcite-panel/calcite-panel.tsx#L394) set on it, which places it in the focus order. This should be removed.

When inactive elements are allowed to receive keyboard focus, unnecessary tab stops are added for sighted keyboard-only users, and screen reader users may assume that the elements are interactive when they are not.

### Related Standards

- **Avoid placing inactive elements in the focus order:** 
https://amp.levelaccess.net/public/standards/view_best_practice.php?violation_id=524
- https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex#accessibility_concerns

### Reproduction Steps

```html
<calcite-panel>
  <p>Slotted content</p>
</calcite-panel>
```
[Storybook Iframe URL](https://esri.github.io/calcite-components/iframe.html?id=components-panel--basic&args=&knob-dragEnabled=true&knob-multiple=true&knob-selection-follows-focus=true&knob-filterEnabled=true&viewMode=story) (in Chrome, if you use "Tab" key to get to the section you'll see a browser-blue outline.)

### Expected Behavior

[Recommendation] 
Developers should remove the tabindex="0" from the mentioned content so that it does not receive keyboard focus.

[Code Example]
<section class="content-container"><slot></slot></section>

### Remediation Steps

Remove tabindex on the section tag

### Screenshots/Images

![image](https://user-images.githubusercontent.com/42423180/139350942-afde8fbc-9ad4-4b31-8e64-74c673783800.png)

### Additional Context

n/a

### Issue Source
- Instance: https://amp.levelaccess.net/public/reporting/view_instance.php?instance_id=1900480082

**Comment by driskull** [comment]
@caripizza the reason for the tabindex is because when the panel is scrollable, it needs to be able to receive focus. It will fail to be scrollable by keyboard if we remove the tabindex.

**Comment by driskull** [comment]
Same as this issue: https://github.com/Esri/calcite-components/issues/1642

https://dequeuniversity.com/rules/axe/4.0/scrollable-region-focusable

**Comment by driskull** [comment]
This has come up a few times so maybe we need to add a comment or something to the code.

**Comment by caripizza** [comment]
@driskull are you saying this is not a valid Level Access issue and we can close it?

**Comment by jcfranco** [comment]
@driskull Thanks for clarifying. As a (separate) enhancement, could we check if the content has a scrollbar and in that case make the area focusable?

**Comment by driskull** [comment]
>  could we check if the content has a scrollbar and in that case make the area focusable?

I can't think of a great way to do that but that would work.

**Comment by driskull** [comment]
We've talked to level access about this before and they recommended to have it focusable.

**Comment by driskull** [comment]
I think this one is blocked. Should it be in a milestone still?

**Comment by benelan** [comment]
I'll move it into the back burner for now.

**Comment by jcfranco** [comment]
> We've talked to level access about this before and they recommended to have it focusable.

I think we can close if LA suggested this approach. Can you post their guidance here?
